### PR TITLE
Restore Wallet - Recovery phrase step

### PR DESCRIPTION
**Pull request by vikas-cis**
Restore Wallet - Recovery phrase step
Task : https://particl.atlassian.net/browse/PAR-215

### CODE REVIEW

File: src/app/modals/createwallet/passphrase/passphrase.component.html [hunk 1]
```diff
@@ -24,7 +24,7 @@
                  [cbContent]="words.join(' ')">
         </md-icon>
         <md-icon class="cursor-pointer icon" *ngIf="!generate"
-                 mdTooltip="Paste" fontSet="faIcon" fontIcon="fa-clipboard">
+                 mdTooltip="Paste" fontSet="faIcon" fontIcon="fa-clipboard" (mousedown)="pasteContent()">
```
Comment (rynomster) [inline src/app/modals/createwallet/passphrase/passphrase.component.html]: Change to `(click)`

File: src/app/modals/createwallet/passphrase/passphrase.component.ts [hunk 2]
```diff
@@ -84,4 +85,9 @@ export class PassphraseComponent implements  OnChanges {
   copyToClipBoard(): void {
     this.flashNotificationService.open('Wallet recovery phrase copied to clipboard.', '', 1000);
   }
+
+  pasteContent() {
+    document.execCommand('Paste');
+    return false;
```
Comment (rynomster) [inline src/app/modals/createwallet/passphrase/passphrase.component.ts]: Remove return statement

File: src/app/modals/createwallet/passphrase/passphrase.component.ts [hunk 3]
```diff
@@ -42,13 +42,15 @@ export class PassphraseComponent implements  OnChanges {
   }
 
   checkFocus(event: KeyboardEvent, index: number) {
+    console.log(index);
```
Comment (kewde) [inline src/app/modals/createwallet/passphrase/passphrase.component.ts]: remove logging

File: src/app/modals/createwallet/passphrase/passphrase.component.ts [hunk 4]
```diff
@@ -57,6 +59,7 @@ export class PassphraseComponent implements  OnChanges {
     words.forEach((word, i) => {
       if (i + index < MAX_WORDS) {
         this.words[i + index] = this.validateWord(word, -1) ? word : 'INVALID';
+        console.log(this.words[i + index]);
```
Comment (kewde) [inline src/app/modals/createwallet/passphrase/passphrase.component.ts]: rm logging

File: src/app/modals/createwallet/passphrase/passphrase.component.html [hunk 5]
```diff
@@ -21,9 +21,9 @@
       <md-icon fontSet="ncIcon" fontIcon="nc-copy-2"></md-icon> Copy to Clipboard
     </button>
 
-    <button md-button *ngIf="!generate" class="paste TODO">
+    <button md-button *ngIf="!generate" class="paste TODO" (click)="pasteContent()">
```
Comment (kewde) [inline src/app/modals/createwallet/passphrase/passphrase.component.html]: remove todo

**Comment by kewde** [comment]
What does this PR do? I see a lot of focus stuff, but I'm not sure what exactly it is trying to fix?

File: src/app/modals/createwallet/passphrase/passphrase.component.html (hunk 5 above)
Comment (vikas-cis) [reply src/app/modals/createwallet/passphrase/passphrase.component.html]: ok

Comment (vikas-cis) [reply src/app/modals/createwallet/passphrase/passphrase.component.html]: I have updated the PR description and shared the url of the task.

**Comment by kewde** [comment]
```
When all the words are copied and pasted, the words are correctly allocated to the correct spaces (Split up correctly). This functionality should only work when a user paste the recovery phrase in the first box, should the user try to paste from the in the second box and onward, the paste should only apply to that specific box regardless of spaces or length.
```
TODO:

- [ ] Test functionality

**Comment by rynomster** [comment]
I don't know, you might want to paste the last 5 words

**Comment by HenkSwardt** [comment]
The comment above from @kewde was from the original task. https://particl.atlassian.net/browse/PAR-215. This was logged from the UAT document as discussed with Francois.

**Comment by rynomster** [comment]
Yeah, we logged it, but then thought again, that someone might want to paste from anywhere. I'm going to review this in a bit

**Comment by HenkSwardt** [comment]
We will remove the restriction added that stopped you from Pasting the string of words in any other box than the first one.

Allowing paste and split in any box

**Comment by allienworks** [comment]
Shouldn't we close this ancient PR and start again from scratch on `dev` branch?

**Pull request closed without merging**